sklearn/tests/test_base.py

```diff
@@ -509,3 +516,64 @@ def test_regressormixin_score_multioutput():
            "built-in scorer 'r2' uses "
            "multioutput='uniform_average').")
     assert_warns_message(FutureWarning, msg, reg.score, X, y)
+
+
+def test_validate_X():
+    # Make sure ValueError is raised when there is a n_features mismatch
+    # between fit and predict/transform
+
+    X = [[0, 1], [2, 3]]
+
+    ss = StandardScaler().fit(X)
+    ss.transform(X)  # All good
+
+    with pytest.raises(ValueError, match="X has 3 features, but"):
+        X_more_features = [[0, 1, 4], [2, 3, 5]]
+        ss.transform(X_more_features)
+
+
+def test_n_features_in_attribute():
+    # Make sure n_features_in_ is correctly set.
```

Comment:
If we are going with a public attribute should this be in estimator_checks??

Reply:
it definitely should!